### DISCUSSION TITLE
Fortio UI not remembering headers

### DISCUSSION THREAD

**Discussion by MandarJKulkarni** [fortio/fortio#734] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/33712629/225915603-84b8bcba-e896-439c-a9b9-c8193057679a.png)

I am passing 3 headers in total for a load test.
Once the execution is complete, I navigate back to the main page and only one header is remembered.

![image](https://user-images.githubusercontent.com/33712629/225915798-a3fe2ae7-4829-43ff-90a2-6b4bc5e6933c.png)

How can I make sure that UI remembers all the headers?

Also for the executions done through UI, how can I get an equivalent command-line?
In some cases, it is quite cumbersome to generate the command line when payload is large.

**Comment by ldemailly** [comment]
The command line is easy, it's the same argument names and you can also just copy paste the url as well as use `/fortio/rest/run` which accepts the same arguments and also can take them from a json payload (some example of that in the README)

For the "back" button, it works for me on Chrome Version 111.0.5563.64 (Official Build) (arm64) on mac

What browser are you using?

<img width="188" alt="Screen Shot 2023-03-17 at 6 47 10 PM" src="https://user-images.githubusercontent.com/3664595/226076901-13d3e477-ed05-4b0a-980a-d65837204800.png">

**Reply by ldemailly** [reply]
also works on safari and also works on firefox for me...

**Reply by MandarJKulkarni** [reply]
Chrome is up to date
Version 111.0.5563.65 (Official Build) (64-bit)

I have this version. It is not working for me on Windows 10.

**Reply by MandarJKulkarni** [reply]
Also why I wanted to get a command line equivalent of the UI is that the header I am passing is quite big and when I pass that header in the UI it works, but when I pass the same string using -H argument to fortio load, I get error like "At most 1 argument expected, got 16" and I am clueless as to what is wrong. It took me a while to figure out which header could be problematic.
The examples in the readme are quite basic to be honest, just foo:bar examples. My suggestion would be, if I may, to provide some help regarding escaping special characters in headers or payload, if we need to URL encode anything etc. 
Of course I appreciate the quick responses here.

**Reply by ldemailly** [reply]
That's a shell question, and you just need to quote your header properly

ie -H "name: value..." if your value contains " then use ' if it contains both then you need to escape ie

-H "name: value \\" 'rest"

**Reply by ldemailly** [reply]
Also (still a shell question) 

-H "name: $(cat value.txt)"

with anything in value.txt file

maybe you need to use wsl or git bash or google how to escape/quote arguments in windows native cmd shell

**Reply by MandarJKulkarni** [reply]
Thank you very much for the replies. The point I am trying to make is, yes I can figure out certain things with help of Google, but the fortio UI already does the proper conversions while passing headers and payload to backend server. If I could just get the proper command line or curl in the output json (or on command prompt where I've launched server) for what I am running through UI, it would save me some time. Please let me know if I am missing something.

**Reply by ldemailly** [reply]
you can copy the url from your browser after clicking start and that url can be curl'ed as many times as you want, each time it will start a similar run

**Reply by ldemailly** [reply, marked as answer]
You can also use the rapi endpoint `/fortio/rest/run` which take similar url or json parameters

Though doc could be better #662
